forem/forem
# Static Sections for Site Navigation - Pt1

## What type of PR is this? (check all applicable)


## Description
This PR represents the first part of implementing the [Static Sections for Nav Links RFC](https://github.com/forem/rfcs/blob/2b110df1a11942d0b83126024055aba1cd7b9fed/text/0237-static-sections-nav-links.md). In this PR, I completed the following:
- [X] Add column ‘section’ to navigation_links table. Make it an enum: 0 = default; 1 = other.
- [X] Create DataUpdateScript that updates existing links if they should be in ‘Other Section’
- [X] Update seeds and associated rake tasks
- [ ] ~Add scopes for Default Links and Other Links~ (not necessary)
- [X] Write and/or fix related specs

Part 2 of my implementation will complete necessary changes to both the frontend and admin section. This PR does NOT change the NavigationLink creation-flow, or display.

## QA Instructions, Screenshots, Recordings
1. Check that you have some navigation links in your database: `bin/rails console; NavigationLink.count`.
2. Run migrations (`bin/rails db:migrate`). In Rails Console, check that the value for `section` on all Navigation Links is `"default"`.
3. Run data-update scripts (`bin/rails data_updates:run`). Now in Rails Console, check that the `section` value for `"/privacy"`, `"/terms"` and `"/code-of-conduct"` navigation links are `"other"`.
4. Reset your database (`bin/rails db:reset`). In Rails Console, check that the navigation links `"/privacy"`, `"/terms"` and `"/code-of-conduct"` already have `section: "other"`.
5. Run all specs and ensure they pass.

### UI accessibility concerns?
None, no front-end changes

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
- [X] I will share this change internally with the appropriate teams. **Eventually, when the entire RFC is implemented, I will update the documentation and liaise with the Community Team to add to our Change Log post.**
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

![Eddie Murphy sipping champagne](https://media.giphy.com/media/5xtDarqlsEW6F7F14Fq/giphy-downsized.gif)
